src/live_restore/live_restore_fs.c

```diff
@@ -804,57 +667,109 @@ __live_restore_fh_read(
      *
      * Right now reads and writes are atomic if we unlock early we lose some guarantee of atomicity.
      */
-    __wt_readunlock(session, &lr_fh->ext_lock);
+    __wt_readunlock(session, &lr_fh->bitmap_lock);
 
     return (ret);
 }
 
+/*
+ * __live_restore_fh_size --
+ *     Get the size of a file in bytes, by file handle.
+ */
+static int
+__live_restore_fh_size(WT_FILE_HANDLE *fh, WT_SESSION *wt_session, wt_off_t *sizep)
+{
+    WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh;
+    wt_off_t destination_size;
+
+    lr_fh = (WTI_LIVE_RESTORE_FILE_HANDLE *)fh;
+
+    WT_RET(lr_fh->destination.fh->fh_size(lr_fh->destination.fh, wt_session, &destination_size));
+    *sizep = destination_size;
+    return (0);
+}
+
+/*
+ * __live_restore_compute_read_end_bit --
+ *     Compute the last possible bit for hole filling.
+ */
+static int
+__live_restore_compute_read_end_bit(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh,
+  wt_off_t buf_size, uint64_t first_clear_bit, uint64_t *end_bitp)
+{
+    wt_off_t read_start = WTI_BIT_TO_OFFSET(first_clear_bit);
+    wt_off_t file_size;
+    /*
+     * In theory we have truncated the destination file to be smaller than that of the source file.
```

Comment:
```suggestion
     * In theory we have truncated the destination file to be smaller than the source file.
```

Reply:
https://github.com/wiredtiger/wiredtiger/commit/ba9df39c3aaad23519e208e9807542bbd025dc17

Reply:
sorry I don't see this change

Reply:
My b: 36ec29df1